app-modules/ai/src/Filament/Pages/Assistant/Concerns/CanManageThreads.php

```diff
@@ -213,6 +221,108 @@ public function selectThread(?AiThread $thread): void
+
+        // Currently threads only support a single vector store
+        $expiredVectorStores = collect($thread->vectorStoreIds)
+            ->map(function ($vectorStoreId) {
+                $vectorStoreResponse = $this->thread->assistant->model->getService()->retrieveVectorStore($vectorStoreId);
+
+                if ($vectorStoreResponse->status === 'expired') {
+                    return $vectorStoreResponse;
+                }
+
+                return null;
+            })
+            ->filter()
+            ->toArray();
+
+        return ! empty($expiredVectorStores) ? $expiredVectorStores : null;
+    }
+
+    public function recreateVectorStoreForThread(AiThread $thread, array $vectorStore): void
+    {
+        $vectorStoreFileIds = [];
+
+        $this->retrieveAllVectorStoreFileIds(
+            thread: $thread,
+            vectorStoreId: $vectorStore['id'],
+            vectorStoreFileIds: $vectorStoreFileIds
+        );
+
+        // Create new vector store
+        $newVectorStore = $thread->assistant->model->getService()->createVectorStore([
+            'file_ids' => $vectorStoreFileIds,
+            'name' => 'Refreshed vector store ' . now()->timestamp . ' for thread' . $thread->id,
+        ]);
+
+        // Update the thread to use the new vector store.
+        $thread->assistant->model->getService()->modifyThread($thread, [
+            'tool_resources' => [
+                'file_search' => [
+                    'vector_store_ids' => [$newVectorStore->id],
+                ],
+            ],
+        ]);
+
+        // Ensure the new vector store has processed all of its files.
+        $this->awaitVectorStoreProcessing(
+            thread: $thread,
+            vectorStore: $newVectorStore
+        );
+    }
+
+    public function retrieveAllVectorStoreFileIds($thread, $vectorStoreId, &$vectorStoreFileIds = [], $after = null)
+    {
+        $params = [];
+
+        if ($after !== null) {
+            $params['after'] = $after;
+        }
+
+        $response = $thread->assistant->model->getService()->retrieveVectorStoreFiles($thread, $vectorStoreId, $params);
+
+        collect($response->data)->each(function ($file) use (&$vectorStoreFileIds) {
+            $vectorStoreFileIds[] = $file['id'];
+        });
+
+        if ($response->hasMore === true) {
+            $this->retrieveAllVectorStoreFileIds(
+                thread: $thread,
+                vectorStoreId: $vectorStoreId,
+                vectorStoreFileIds: $vectorStoreFileIds,
+                after: $response->lastId
+            );
+        }
+    }
+
+    public function awaitVectorStoreProcessing(AiThread $thread, VectorStoresDataTransferObject $vectorStore): void
+    {
+        $timeout = 60;
+
+        $vectorStoreResponseStatus = $vectorStore->status;
+
+        while ($vectorStoreResponseStatus !== 'completed') {
+            if ($timeout <= 0) {
+                throw new AiThreadLockedException();
```

Comment:
I think we will want to throw an explicit message here that can describe to the User that it is processing files in the thread and to please wait or something like that.

It will also allow us to better see what errors are being thrown in error reporting.